internal/services/devcenter/dev_center_network_connection_resource_test.go

```diff
@@ -0,0 +1,242 @@
+  resource_group_name = azurerm_dev_center_network_connection.test.resource_group_name
+  location            = azurerm_dev_center_network_connection.test.location
+  domain_join_type    = azurerm_dev_center_network_connection.test.domain_join_type
+  subnet_id           = azurerm_dev_center_network_connection.test.subnet_id
+}
+`, r.basic(data))
+}
+
+func (r DevCenterNetworkConnectionTestResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "network" {
+  name     = "acctestrg-dcncn-%d"
+  location = "%s"
+}
+
+resource "azurerm_dev_center_network_connection" "test" {
+  name                = "acctest-dcnc-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  domain_join_type    = "HybridAzureADJoin"
+  subnet_id           = azurerm_subnet.test.id
+  domain_name         = "never.gonna.shut.you.down"
+  domain_username     = "tfuser@microsoft.com"
+  domain_password     = "P@ssW0RD7890"
+  organization_unit   = "OU=Sales,DC=Fabrikam,DC=com"
+
+  tags = {
+    ENV = "Test"
+  }
+}
+`, r.template(data), data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func (r DevCenterNetworkConnectionTestResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "network" {
+  name     = "acctestrg-dcncn-%d"
+  location = "%s"
+}
+
+resource "azurerm_subnet" "test2" {
+  name                 = "internal2"
+  resource_group_name  = azurerm_resource_group.test.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefixes     = ["10.0.3.0/24"]
+}
+
+resource "azurerm_dev_center_network_connection" "test" {
+  name                = "acctest-dcnc-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  domain_join_type    = "HybridAzureADJoin"
+  subnet_id           = azurerm_subnet.test2.id
+  domain_name         = "never2.gonna.shut.you.down"
+  domain_username     = "tfuser2@microsoft.com"
+  domain_password     = "P@ssW0RD7891"
+  organization_unit   = "OU=SaleStores,DC=Fabrikam,DC=com"
+
+  tags = {
+    ENV = "Test2"
+  }
+}
+`, r.template(data), data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func (r DevCenterNetworkConnectionTestResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestrg-dcnc-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctest-vnet-%d"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  lifecycle {
+    ignore_changes = [subnet]
+  }
```

Comment:
This change has been reverted so we should remove this 
```suggestion
```

Reply:
Updated